app/src/main/java/com/checkin/app/checkin/Home/NearbyRestaurantAdapter.kt

```diff
@@ -1,43 +1,126 @@
 package com.checkin.app.checkin.Home
 
+import android.app.PendingIntent.getActivity
+import android.content.Context
+import android.util.Log
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.ImageView
+import androidx.fragment.app.FragmentActivity
 import androidx.recyclerview.widget.RecyclerView
+import androidx.viewpager.widget.PagerAdapter
+import androidx.viewpager.widget.ViewPager
 import com.checkin.app.checkin.Home.model.NearbyRestaurantModel
 import com.checkin.app.checkin.R
 import com.checkin.app.checkin.Utility.BaseViewHolder
+import com.rd.PageIndicatorView
+import com.rd.animation.type.AnimationType
+import java.net.URL
+import java.util.*
+import android.graphics.BitmapFactory
+import android.graphics.Bitmap
+import android.nfc.Tag
+
+
+class NearbyRestaurantAdapter(var context: Context) : RecyclerView.Adapter<BaseViewHolder<Any>>() {
+
+
 
-class NearbyRestaurantAdapter : RecyclerView.Adapter<BaseViewHolder<Any>>() {
     var data: List<NearbyRestaurantModel> = emptyList()
 
     fun updateData(newData: List<NearbyRestaurantModel>) {
+
         data = newData
         notifyDataSetChanged()
     }
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BaseViewHolder<Any> {
         val view = LayoutInflater.from(parent.context).inflate(viewType, parent, false)
-        return if (viewType == R.layout.item_home_restaurant_banner) NearbyRestaurantViewHolder(view) as BaseViewHolder<Any> else AdvertisementViewHolder(view)
+        return if (viewType == R.layout.item_home_restaurant_banner) NearbyRestaurantViewHolder(view)
+                as BaseViewHolder<Any> else AdvertisementViewHolder(view)
     }
 
-    override fun getItemCount(): Int = data.count() + 1
+    override fun getItemCount(): Int = data.count()
 
-    override fun getItemViewType(position: Int): Int = if (position == 2) R.layout.item_advertisement_banner else R.layout.item_home_restaurant_banner
+    override fun getItemViewType(position: Int): Int = if (position == 2)
+        R.layout.item_advertisement_banner else R.layout.item_home_restaurant_banner
 
     override fun onBindViewHolder(holder: BaseViewHolder<Any>, position: Int) {
         holder.bindData(data[position])
     }
 
     inner class NearbyRestaurantViewHolder(itemView: View) : BaseViewHolder<NearbyRestaurantModel>(itemView) {
         override fun bindData(data: NearbyRestaurantModel) {
-            TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+            var imageView:ImageView=itemView.findViewById(R.id.home_image)
```

Comment:
Butterknife can be used inside the ViewHolder. In the init block of viewholder, just use 
`Butterknife.bind(this, itemView);`
For some code example, check out other viewholders in app

Reply:
It's actually inefficient to call findViews in bindData since it is a slow operation and needn't be computed again and again for binding data. bindData should mostly concern itself with binding the data to UI, not related to how UI flow goes (like click listeners)

Reply:
Makes perfect sense, working to fix these and few others I spotted